add tljs wrapper

y21 opened the pull request:
This PR adds [@y21/tljs](https://npmjs.com/@y21/tljs) to the list of parsers.

Old description (no longer relevant): ~~This package makes use of WebAssembly, so it must compile the module before anything can be parsed.
Of course this doesn't count against actual parsing time so I've made it possible to export a `setup` function that is run once before benchmarking. In the case of tljs it instantiates the module once and caches it, which means all subsequent parse calls will reuse this module.~~

AndreasMadsen commented:
Well, we can add this feature if necessary. But you should consider synchronous initialization for node.js, see for example https://github.com/nearform/node-cephes/blob/master/cephes-wrapper.js#L91 . As far as I'm aware, there are no size limitation on the WASM filesize in node.js.

AndreasMadsen commented on wrapper/tl.js:
```diff
@@ -0,0 +1,12 @@
+module.exports = function(html, callback) {
+    tljs.parse(html).then((dom) => {
+        callback();
+        dom.free();
```
I'm not convinced having a free call after `callback` isn't cheating. Either move it to before or implement `WeakRef` in `@y21/tljs`. You should properly implement `WeakRef` no matter what to avoid a memory leak.

y21 commented:
So, do you suggest I keep the part that checks if the module contains a setup function (if it's ever needed for other implementations in the future), or remove it? As for tljs, I agree with your comments. A synchronous function for initializing should exist, and wasm resources should make use of `WeakRef`/`FinalizationRegistry` to free resources when their object is GC'd instead of having to manually call free.

AndreasMadsen commented:
Let's not add features (setup support) that are not needed.

y21 commented:
Alright, I reverted the setup stuff and removed the free call. `@y21/tljs` uses a global FinalizationRegistry for DOM handles now, that when the handle to a DOM object is GC'd, will deallocate all associated resources in WebAssembly memory.

AndreasMadsen commented on _run.js:
```diff
@@ -28,4 +28,4 @@ process.on('message', function (item) {
 		process.exit(0);
 	});
-});
+});
```
please leave the newline

AndreasMadsen commented on wrapper/tl.js:
```diff
@@ -0,0 +1,6 @@
+tljs.initializeWasmSync();
+
+module.exports = function(html, callback) {
+    tljs.parse(html).then(() => callback());
```
Please relay any potential error to the callback. You could use `util.callbackify`.

y21 commented:
Added the newline back (not sure why my editor removed them in the first place) and callback-ified the parse function.

AndreasMadsen commented:
Great. I have merged it.
